p2p/exceptions.py

```diff
@@ -159,3 +159,10 @@ class NoInternalAddressMatchesDevice(BaseP2PError):
     def __init__(self, *args: Any, device_hostname: str=None, **kwargs: Any) -> None:
         super().__init__(*args, **kwargs)
         self.device_hostname = device_hostname
+
+
+class ValidationError(BaseP2PError):
```

Comment:
I'm a bit worried about this as it's the same name of the exception from the `eth` package, and I believe we'll often end up writing code that catches one when in fact is the other that will be raised. Like we just saw with the `OperationCancelled` exception after the move to the cancel_token lib

Reply:
cc @carver 

how would the two of you feel about a common/shared validation library that exposed a common `ValidationError` exception that we could use across all of our libraries.  It is one of those things that we have implemented almost everywhere.

Reply:
https://github.com/ethereum/py-evm/issues/1125